doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,295 @@
+show, where adding additional functionality to the StandardScaler class adds
+the pandas in pandas out capability. Care should be taken as this does not
+take care of the column ordering problem that is discussed in the next section.
+
+The column ordering problem
+===========================
+
+Because Scikit-learn transforms DataFrames to NumPy arrays, it should be
+assumed, that all information and benefits of column names is lost and that
+from that point forward, only column order and not column labels stay relevant.
+This can cause problems in general when predicting unseen data using a previously
+trained estimator and applying it to the new data as it does not matter
+that the unseen/new data has the same data columns and labels, they still
+**must** be provided in the correct order too.
+Scikit-learn does not check that the column order is consistent nor does
+it do any automatic re-ordering of DataFrame columns!
+
+An example of how this might impact your future prediction can be seen in the
+example given below::
+
+  >>> from sklearn.datasets import load_iris
+  >>> import pandas as pd
+  >>>
+  >>> # make a dataframe
+  >>> iris = load_iris()
+  >>> X, y = iris.data[:-1,:], iris.target[:-1]
+  >>> iris_pd = pd.DataFrame(X)
+  >>> iris_pd.columns = iris.feature_names
+  >>> iris_pd['target'] = y
+  >>>
+  >>> from sklearn.model_selection import train_test_split
+  >>> train, test = train_test_split(iris_pd, test_size= 0.3, random_state=42)
+  >>>
+  >>> feature_columns_train = ['sepal length (cm)','sepal width (cm)',
+  ...                          'petal length (cm)','petal width (cm)']
+  >>> # last two correct order
+  >>> feature_columns_test = ['sepal length (cm)','sepal width (cm)',
+  ...                         'petal width (cm)','petal length (cm)']
+  >>> # last two switched order
+  >>>
+  >>> from sklearn.linear_model import LogisticRegression
+  >>> lg = LogisticRegression(n_jobs=4, random_state=123, verbose=0,
+  ...                         penalty='l2', C=1.0,
+  ...                         solver='lbfgs', multi_class='auto')
+  >>> lg.fit(train[feature_columns_train], train['target'])
+  LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
+                     intercept_scaling=1, l1_ratio=None, max_iter=100,
+                     multi_class='auto', n_jobs=4, penalty='l2', random_state=123,
+                     solver='lbfgs', tol=0.0001, verbose=0, warm_start=False)
+  >>>
+  >>> res1 = lg.predict(test[feature_columns_train])
+  >>> res1[:5]
+  array([1, 0, 2, 1, 1])
+  >>> # result is actually
+  >>> res2 = lg.predict(test[feature_columns_test])
+  >>> res2[:5]
+  array([0, 0, 2, 0, 0])
+
+
+At the time of writing, it is the users responsibility to ensure that the
+column ordering in the data used for training the estimator is the same as the
+ordering of the data used for prediction. There is an ongoing discussion
+whether or not this will change in the future and this
```

Comment:
I think the only likelihood in the future is that where it would otherwise be passed to asarray and its column names mismatch those used at fit, this will lead to an error. You shouldn't give the impression that we might support automatically reordering columns. This is a very unlikely (and risky) behaviour for us to support.